skyportal/tests/api/test_comments.py

```diff
@@ -12,3 +12,11 @@ def test_add_and_retrieve_comment(comment_token, public_source):
 
     assert status == 200
     assert data['data']['comment']['text'] == 'Comment text'
+
+
+def test_cannot_add_comment_without_permission(view_only_token, public_source):
+    status, data = api('POST', 'comment', data={'source_id': public_source.id,
+                                                'text': 'Comment text'},
+                       token=view_only_token)
+    assert status == 400
```

Comment:
Minor issue for later: would be nice if we could return a 403 instead.  See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status